osbs/repo_utils.py

```diff
@@ -59,6 +61,26 @@ def __init__(self, dir_path='', file_name=REPO_CONFIG_FILE):
         if os.path.exists(file_path):
             with open(file_path) as f:
                 self.container = (yaml.load(f) or {})
+        self._validate_container_modules_nsv()
+
+    def _validate_container_modules_nsv(self):
+        modules = self.container.get('compose', {}).get('modules', [])
+        for module in modules:
+            pieces = module.split(':')
+            if not 1 < len(pieces) < 4:
+                raise OsbsValidationException(
+                    'Module specification should be NAME:STREAM or NAME:STREAM:VERSION')
+
+    def get_container_module_specs(self):
+        ModuleSpec = namedtuple('ModuleSpec', 'name, stream, version')
+        ModuleSpec.__new__.__defaults__ = (None,) * len(ModuleSpec._fields)
```

Comment:
Don't we only want a single default here? `(None,)`, not `(None,) * len(ModuleSpec._fields)`? The name and stream are required I think.

Could add a comment here like `# For compatibility with Python < 3.7` so we can tidy this up to use the nicer 'defaults' parameter in future.